rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -466,11 +466,11 @@ where
                                 Ok(()) => {
                                     log::debug!("Device Update Notification dispatched for device {device_name:?} and inbound endpoint {inbound_endpoint_name:?}");
                                 }
-                                Err(DispatchError::SendError(tokio::sync::mpsc::error::SendError((payload, _)))) => {
-                                    log::warn!("Device Update Observation has been dropped. Received Device Update Notification: {payload:#?}");
+                                Err(DispatchError { data, kind: DispatchErrorKind::SendError }) => {
+                                    log::warn!("Device Update Observation has been dropped. Received Device Update Notification: {data:#?}");
```

Comment:
small nit that's been driving me crazy when debugging, can you change this to not be a pretty print anymore while you're in here?
```suggestion
                                    log::warn!("Device Update Observation has been dropped. Received Device Update Notification: {data:?}");
```